packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2533,7 +2543,24 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     if (shouldShowCaret) {
       _scheduleShowCaretOnScreen();
     }
-    _formatAndSetValue(value, cause, userInteraction: true);
+
+    final TextEditingValue formattedValue = _formatUserInput(value);
+    if (value.selection != _value.selection) {
+      requestKeyboard();
+    }
+
+    if (value.text != _value.text || value.composing != _value.composing) {
+      hideToolbar();
+      _currentPromptRectRange = null;
+
+      if (_hasInputConnection) {
+        if (widget.obscureText && value.text.length == _value.text.length + 1) {
```

Comment:
I got really worried about extended grapheme clusters and obscured fields when I saw this `+ 1` here (even though I know it has nothing to do with this PR).  Indeed they show up as multiple characters when entered into an obscured field and don't have a delay, but it turns out that's the behavior on native too 🤷 .  All is well.